### PR TITLE
feat: add form field id validation

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Implements task "Forms: Add a validator to field id"

## How Has This Been Tested?
Manual

## Documentation
None

### CODE REVIEW

File: packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx [hunk 1]
```diff
@@ -32,6 +33,20 @@ const GeneralTab: React.FC<GeneralTabProps> = ({ field, form }) => {
         throw new Error("Please enter a unique Field ID");
     }, []);
 
+    const fieldIdValidator: Validator = (value: string | undefined) => {
```
Comment (adrians5j) [inline packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx]: 1. Let's move these two validators outside of the `GeneralTab` component. 
2. Let's assign `Validator` type to `uniqueFieldIdValidator`, the same way you did with `fieldIdValidator`. I know it's not part of your changes, still, let's sync it up.

File: packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx [hunk 2]
```diff
@@ -32,6 +33,20 @@ const GeneralTab: React.FC<GeneralTabProps> = ({ field, form }) => {
...
+        if (!value) {
+            return true;
+        }
+
+        if (/^[a-zA-Z0-9_-]*$/.test(value)) {
+            return true;
+        }
+
+        throw Error(
+            'Field ID can only contain letters, numbers and chars ("-", "_") without any spaces.'
```
Comment (adrians5j) [inline packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx]:
```suggestion
            "Field ID may contain only letters, numbers and "-" and "_" characters."
```

**Review by adrians5j — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor changes.

File: packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx (hunk 1 above)
Comment (neatbyte-vnobis) [reply packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx]: @adrians5j 
please add more details for a item 2) about `uniqueFieldIdValidator`.
It just that this validator is using not only a field value, but also uses `useFormEditor` hook to check for ID uniqueness.
So if it is outside a component with type `Validator` I have a difficulty to pass `useFormEditor` value in some way due to type restriction.

Comment (adrians5j) [reply packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx]: I hear you @neatbyte-vnobis, I missed the `useFormEditor` fact.

OK, let's then just use the `useCallback` hook when defining the `fieldIdValidator`, the same way it's done with `uniqueFieldIdValidator`. Also, see if the `Validator` can be used with the `uniqueFieldIdValidator`, again just for consistency.

**Comment by neatbyte-vnobis** [comment]
@adrians5j 
Just in case, you have messaged in Slack "provided an alternative, after that we'll be good to merge".
Have I missed some code sample? Or I misunderstood and we can merged this pull as is and "alternative" is a matter of future?

**Comment by neatbyte-vnobis** [comment]
@adrians5j please take a look on comment above

**Comment by adrians5j** [comment]
Hey @neatbyte-vnobis , sorry for the fuss, by "alternative", I meant:

> OK, let's then just use the useCallback hook when defining the fieldIdValidator, the same way it's done with uniqueFieldIdValidator. Also, see if the Validator can be used with the uniqueFieldIdValidator, again just for consistency.

**Comment by neatbyte-vnobis** [comment]
@adrians5j got it
Please take  a look on updated PR

**Review by adrians5j — APPROVED** [review]
Thank you @neatbyte-vnobis !

**Pull request merged**